### PR TITLE
skesa 2.4.0

### PR DESCRIPTION
**Pull request by tseemann**
- [ ] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

### CODE REVIEW

**Comment by sjackman** [comment]
```
c++ -std=c++11 -c -o kmercounter.o kmercounter.cpp -D NO_NGS -Wall -Wno-format-y2k  -pthread -fPIC -O3 -finline-functions -fstrict-aliasing -fomit-frame-pointer -msse4.2 -I /usr/local/opt/boost/include 
skesa.cpp:31:10: fatal error: malloc.h: No such file or directory
```
https://github.com/brewsci/homebrew-bio/pull/951/checks?check_run_id=499742189#step:4:311
The header for the `malloc` function is `cstdlib` rather than `malloc.h`.
See https://en.cppreference.com/w/cpp/memory/c/malloc

**Comment by tseemann** [comment]
Yeah i fixed that but it still won't build. Issues with boost linking on mac

**Comment by sjackman** [comment]
Boost is compiled with Apple `clang` on macOS, so you probably can't compile Skesa with GNU GCC. I don't believe GNU GCC and Clang are ABI compatible.

File: Formula/skesa.rb [hunk 1]
```diff
@@ -13,21 +12,29 @@ class Skesa < Formula
     sha256 "90e08e2a0fbf08060b2ffbbd3e3951840af8fd03e1fea9113758fe67677dc4d1" => :x86_64_linux
   end
 
+  depends_on "gcc" => :build
```
Comment (sjackman) [inline Formula/skesa.rb]:
```suggestion
```

File: Formula/skesa.rb [hunk 2]
```diff
@@ -13,21 +12,29 @@ class Skesa < Formula
...
+
   depends_on "boost"
 
   uses_from_macos "zlib"
 
+  fails_with :clang
```
Comment (sjackman) [inline Formula/skesa.rb]:
```suggestion
```

File: Formula/skesa.rb [hunk 3]
```diff
@@ -13,21 +12,29 @@ class Skesa < Formula
...
+
   def install
     makefile = "Makefile.nongs"
 
     # https://github.com/ncbi/SKESA/issues/6
     if OS.mac?
       inreplace makefile, "-Wl,-Bstatic", ""
       inreplace makefile, "-Wl,-Bdynamic -lrt", ""
+      inreplace "skesa.cpp", "malloc.h", "stdlib.h"
```
Comment (sjackman) [inline Formula/skesa.rb]:
```suggestion
      inreplace "skesa.cpp", "malloc.h", "cstdlib"
```

File: Formula/skesa.rb [hunk 4]
```diff
@@ -13,21 +12,29 @@ class Skesa < Formula
...
     end
 
-    system "make", "-f", makefile, "BOOST_PATH=#{Formula["boost"].opt_prefix}"
-    bin.install "skesa"
+    ENV.cxx11
```
Comment (sjackman) [inline Formula/skesa.rb]:
```suggestion
```

File: Formula/skesa.rb [hunk 5]
```diff
@@ -13,21 +12,29 @@ class Skesa < Formula
...
+    system "make", "-f", makefile,
+      "CC=#{ENV.cxx}",
```
Comment (sjackman) [inline Formula/skesa.rb]:
```suggestion
```

**Comment by tseemann** [comment]
@sjackman Alex has made some changes to HEAD which might help
https://github.com/ncbi/SKESA/compare/2.4.0...master
I don't have access to MacOS for 24 hours.

**Pull request merged**